File: blogExamples/datatypes/arrays/arrayEachSux/ruby/sparse.rb
rainbow = []
rainbow[0] = "Whero"
rainbow[2] = "Kowhai"
rainbow[4] = "Kikorangi"
rainbow[6] = "Mawhero"

callCounter = 0
rainbow.each do |v|
		callCounter += 1
		puts("Called for #{v}")
end
puts("callback was called: #{callCounter} times")

File: languageComparison/arrayEach/js/sparse.js
rainbow = [];
rainbow[1] = "Whero";
// no [2]
rainbow[3] = "Kowhai";
// no [4]
rainbow[5] = "Kikorangi";
// no [6]
rainbow[7] = "Mawhero";

callCounter = 0;
rainbow.forEach(
	function(v,i,a){
		callCounter++;
		console.log("Called for [" + i + "]: " + v);
	}
);
console.log("callback was called: " + callCounter + " times");